app/model.js

```diff
@@ -16,8 +17,18 @@ exports.selectSpecifiedArticle = (id)=>{
     return db.query(`SELECT * FROM articles WHERE article_id = $1`,[id])
 }
 
-exports.selectAllArticles = ()=>{    
-    return db.query(`SELECT articles.article_id, title, topic, articles.author, articles.created_at, articles.votes, article_img_url, count(*) AS comment_count FROM articles JOIN comments ON comments.article_id = articles.article_id GROUP BY articles.article_id ORDER BY articles.created_at DESC`)
+exports.selectAllArticles = (query)=>{    
+    let querystr = `SELECT articles.article_id, title, topic, articles.author, articles.created_at, articles.votes, article_img_url, count(*) AS comment_count FROM articles
+     JOIN comments ON comments.article_id = articles.article_id `
+
+    const queryVar = []
+    if(Object.keys(query).length>0){
+        if(query.topic===undefined){throw err}
```

Comment:
throw what err? We might want to be more specific about the error here and consider a test for this scenario if it's one we need.